src/app/hearing-part/components/listing-create/listing-create.component.ts

```diff
@@ -202,36 +241,62 @@ export class ListingCreateComponent implements OnInit {
                 this.listing.hearing.hearingTypeCode,
                 [Validators.required]
             ),
-            duration: new FormControl(
-                this.listing.hearing.duration ? this.listing.hearing.duration.asMinutes() : undefined,
-                [Validators.required, Validators.min(1), Validators.max(this.limitMaxValue)]
+            listingTypeRadio: new FormControl(
+                this.chosenListingType, [Validators.required]
             ),
+            listingType: new FormGroup({
+                duration: new FormControl({
+                        value: this.getDurationToDisplay(),
+                        disabled: this.editMode && this.isMultiSession()
+                    },
+                    [Validators.required, Validators.min(1), Validators.max(this.limitMaxValue)]
+                ),
+                numberOfSessions: new FormControl(
+                    this.listing.hearing.numberOfSessions, [Validators.min(1)]
+                )
+            }),
             targetDates: new FormGroup({
                 targetFrom: new FormControl(this.listing.hearing.scheduleStart),
                 targetTo: new FormControl(this.listing.hearing.scheduleEnd),
             }, this.validateTargetDates)
         });
     }
 
+    private getDurationToDisplay() {
+        if (this.listing.hearing.duration) {
+            if (this.chosenListingType === ListingTypeTab.Multi) {
```

Comment:
This condition can be moved to the one above with &&, or you can maybe use unary operator here.

Reply:
no, because if chosenListingType is not multi we want to return minutes, not undefined;
and if the duration is empty we want to return undefined, not an error;